katran/lib/BalancerStructs.h

```diff
@@ -53,6 +53,16 @@ struct vip_meta {
   uint32_t vip_num;
 };
 
+// real's definition for lookup
+struct real_definition {
+  union {
+    __be32 dst;
+    __be32 dstv6[4];
+  };
+  __u8 flags;
+  __u64 real_flags;  //real flags
+};
+
```

Comment:
1. why not reuse existing flags? (__u8; only 1 is used there to distinct v4 vs v6). also it is already defined in https://github.com/facebookincubator/katran/blob/master/katran/lib/IpHelpers.h#L28 so no need to redifine real_definition

Reply:
To be real's flags as same as vip's flags (__u64), and future use of current __u8 flags, I  thought it is good to have this field in the definition. Anyway, I change the code to use current flags. Thanks for the comment.

Reply:
Could you check lease also rework the code in a way that local ip (up which if real is equal to) would be configured with a function instead of new per real flag. And in updateReal would be compared against. It would simplify the code greatly.

Reply:
Sure. I am currently working on it and trying to test its functionality.